x-pack/plugins/fleet/public/applications/integrations/sections/epm/screens/detail/settings/uninstall_button.tsx

```diff
@@ -34,7 +30,9 @@ export const UninstallButton: React.FunctionComponent<UninstallButtonProps> = ({
   title,
   version,
 }) => {
-  const hasWriteCapabilites = useCapabilities().write;
+  const hasFleetWritePermissions = useAuthz().fleet.all;
+  const hasIntWritePermissions = useAuthz().integrations.installPackages;
+  const hasAllWritePermissions = hasFleetWritePermissions && hasIntWritePermissions;
```

Comment:
We shouldn't do any logic in our routes or UI code and instead implement this in `calculateAuthz` so that we have consistent enforcement across server and client code. Otherwise we're going to get divergence. If an existing key on FleetAuthz doesn't provide what is needed, we should add one.

It seems we already have a `authz.integrations.removePackages` key that does this which I think we can use directly.

Can we update all the UI code to follow this pattern?